feathr_project/feathr/registry/_feature_registry_purview.py

```diff
@@ -966,61 +1146,81 @@ def _list_registered_entities_with_details(self, project_name: str, entity_type:
-        feature_entity_guid_mapping = {x['guid']:x for x in feature_entities}
-
-        # this is guid for feature anchor (GROUP of anchor features)
-        anchor_guid = [anchor_entity["guid"] for anchor_entity in project_entity["attributes"]["anchor_features"]]
-        derived_feature_guid = [derived_feature_entity["guid"] for derived_feature_entity in project_entity["attributes"]["derived_features"]]
-        
-        derived_feature_ids = [feature_entity_guid_mapping[x] for x in derived_feature_guid]
-        
+        entities_dict = {x['guid']:x for x in entities_under_project}
+
         derived_feature_list = []
-        for derived_feature_entity_id in derived_feature_ids:
+
+        for derived_feature_entity in [x for x in entities_under_project if x['typeName']==TYPEDEF_DERIVED_FEATURE]:
             # this will be used to generate DerivedFeature instance
             derived_feature_key_list = []
                 
-            for key in derived_feature_entity_id["attributes"]["key"]:
-                derived_feature_key_list.append(TypedKey(key_column=key["key_column"], key_column_type=key["key_column_type"], full_name=key["full_name"], description=key["description"], key_column_alias=key["key_column_alias"]))
+            for key in derived_feature_entity["attributes"]["key"]:
+                derived_feature_key_list.append(TypedKey(key_column=key["keyColumn"], key_column_type=key["keyColumnType"], full_name=key["fullName"], description=key["description"], key_column_alias=key["keyColumnAlias"]))
             
-            # for feature anchor (GROUP), input features are splitted into input anchor features & input derived features
-            anchor_feature_guid = [e["guid"] for e in derived_feature_entity_id["attributes"]["input_anchor_features"]]
-            derived_feature_guid = [e["guid"] for e in derived_feature_entity_id["attributes"]["input_derived_features"]]
-            # for derived features, search all related input features.
-            input_features_guid = self.search_input_anchor_features(derived_feature_guid,feature_entity_guid_mapping)
-            # chain the input features together
-            # filter out features that is related with this derived feature 
-            all_input_features = self._get_features_by_guid_or_entities(guid_list=input_features_guid+anchor_feature_guid, entity_list=all_entities_in_project) 
-            derived_feature_list.append(DerivedFeature(name=derived_feature_entity_id["attributes"]["name"],
-                                feature_type=self._get_feature_type_from_hocon(derived_feature_entity_id["attributes"]["type"]),
-                                transform=self._get_transformation_from_dict(derived_feature_entity_id["attributes"]['transformation']),
+            def search_for_input_feature(current_bfs_step_entities, full_relations,full_entities):
```

Comment:
Do we still need to do BFS here? Since we are storing all the info in AtlasProcess.

Reply:
Removed recursive call